File: proj/code-repair/mycode/clean_dataset/patch-single-line/quixbugs-python/codex-sh/detect_cycle.py
--- 
+++ 
@@ -2,7 +2,7 @@
     hare = tortoise = node
 
     while True:
-        if hare.successor is None:
+        if hare is None or hare.successor is None:
             return False
 
         tortoise = tortoise.successor

File: proj/code-repair/mycode/clean_dataset/patch-single-line/quixbugs-java/incoder-1/detect_cycle.java
--- 
+++ 
@@ -3,7 +3,7 @@
     Node tortoise = node;
 
     while (true) {
-        if (hare.getSuccessor() == null)
+        if (hare == null || hare.getSuccessor() == null)
             return false;
 
         tortoise = tortoise.getSuccessor();
